### PR TITLE
Introduce a shared pagerduty config for account_name, used to generate api/interface urls

### PR DESCRIPTION
**Pull request by glasnt**
Removes hardcoding raised in issue #16
Helper function in lib/pagerduty.rb allows for controllers and views to access the same value.
Reduced down to an account name for ease of configuration, and more control over the base url (which shouldn't ever have a non pagerduty.com domain name, if I'm not mistaken)

### CODE REVIEW

File: lib/pagerduty.rb [hunk 1]
```diff
@@ -7,11 +7,13 @@ class Pagerduty
   def initialize
     @config = TOML.load_file('config.toml')['pagerduty']
     raise "Could not load credentials file at config.toml" if @config.nil? || @config.empty?
-    # Add trailing slash to API URL if it doesn't already exist
-    @config['api_url'] += '/' unless @config['api_url'].end_with?('/')
     @config['auth_token'] = "Token token=#{@config['auth_token']}" unless @config['auth_token'].start_with?('Token token=')
   end
 
+  def api_url
+    "https://#{@config['account_name']}.pagerduty.com"
```
Comment (Hobbsee) [inline lib/pagerduty.rb]: This should include /api/v1 as well, shouldn't it?

File: lib/pagerduty.rb [hunk 2]
```diff
@@ -40,7 +42,7 @@ def request(endpoint)
   def incidents(start_date, finish_date = nil)
     finish_clause     = finish_date ? finish_clause = "&until=#{finish_date}" : ""
     time_zone = @config['time_zone']
-    endpoint  = "https://bltprf.pagerduty.com/api/v1/incidents?since=#{start_date}#{finish_clause}&time_zone=#{time_zone}"
+    endpoint  = "#{self.api_url}/api/v1/incidents?since=#{start_date}#{finish_clause}&time_zone=#{time_zone}"
```
Comment (Hobbsee) [inline lib/pagerduty.rb]: And then /api/v1 can be removed from here and subsequent calls.

File: views/alert-response.haml [hunk 3]
```diff
@@ -40,7 +40,7 @@
         %tr
           %td= Time.at(incident['alert_time'])
           %td
-            %a{:href => "https://bltprf.pagerduty.com/incidents/#{incident['id']}"}= incident['id']
+            %a{:href => "#{@pagerduty_url}/incidents/#{incident['id']}"}= incident['id']
```
Comment (Hobbsee) [inline views/alert-response.haml]: Although then this would break (unless gsub'd).

**Comment by Hobbsee** [comment]
The cleanest solution is to rename api_url to pagerduty_url, I think.

Looks good otherwise.

**Comment by Hobbsee** [comment]
I made a dog's breakfast of my comments on this PR - sorry about that.

There are two options:
  - Rename api_url to pagerduty_url, which is what it actually is
  - Add /api/v1 to api_url (so it is the actual API URL), and then gsub it out for the GUI.

Out of those, the first option seems like a better plan to me.

**Comment by glasnt** [comment]
I have renamed the function to `pagerduty_url`, so, Option 1.

**Comment by Hobbsee** [comment]
Awesome work, thanks!

**Pull request merged**